analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -387,43 +402,47 @@ public void DeclarationPattern_Var_Deconstruction_DoesNotSetConstraint()
         [DataRow("objectNotNull is not not object", OperationKindEx.TypePattern, true)]
-        [DataRow("objectUnknown is not object", OperationKindEx.TypePattern, null)]
-        [DataRow("objectUnknown is not not object", OperationKindEx.TypePattern, null)]
         [DataRow("exceptionNull is not object", OperationKindEx.TypePattern, true)]
         [DataRow("exceptionNull is not not object", OperationKindEx.TypePattern, false)]
         [DataRow("exceptionNotNull is not object", OperationKindEx.TypePattern, false)]
         [DataRow("exceptionNotNull is not not object", OperationKindEx.TypePattern, true)]
-        [DataRow("exceptionUnknown is not object", OperationKindEx.TypePattern, null)]
-        [DataRow("exceptionUnknown is not not object", OperationKindEx.TypePattern, null)]
-        [DataRow("objectNull is not Exception", OperationKindEx.TypePattern, null)]
-        [DataRow("objectNull is not not Exception", OperationKindEx.TypePattern, null)]
         [DataRow("objectNotNull is not Exception", OperationKindEx.TypePattern, null)]
         [DataRow("objectNotNull is not not Exception", OperationKindEx.TypePattern, null)]
-        [DataRow("objectUnknown is not Exception", OperationKindEx.TypePattern, null)]
-        [DataRow("objectUnknown is not not Exception", OperationKindEx.TypePattern, null)]
         [DataRow("objectNull is not not _", OperationKindEx.DiscardPattern, true)]
         [DataRow("objectNotNull is not not _", OperationKindEx.DiscardPattern, true)]
         [DataRow("objectUnknown is not not _", OperationKindEx.DiscardPattern, null)] // Some patterns always match
-        public void NegateTypeDiscardPatternsSetBoolConstraint(string isPattern, OperationKind expectedOperation, bool? expectedBoolConstraint) =>
+        public void NegateTypeDiscardPatternsSetBoolConstraint_SingleState(string isPattern, OperationKind expectedOperation, bool? expectedBoolConstraint) =>
             ValidateSetBoolConstraint(isPattern, expectedOperation, expectedBoolConstraint);
 
+        [DataTestMethod]
+        [DataRow("objectUnknown", "is not null", OperationKindEx.NegatedPattern, "NotNull", "Null")]
+        [DataRow("objectUnknown", "is not { }", OperationKindEx.NegatedPattern, "Null", "NotNull")]
+        [DataRow("objectUnknown", "is not string { }", OperationKindEx.NegatedPattern, null, "NotNull")]
+        [DataRow("objectUnknown", "is not object { }", OperationKindEx.NegatedPattern, "Null", "NotNull")]
+        [DataRow("objectUnknown", "is not not null", OperationKindEx.NegatedPattern, "Null", "NotNull")]
+        [DataRow("nullableBoolTrue", "is not false", OperationKindEx.NegatedPattern, "True", "False")]
+        [DataRow("nullableBoolFalse", "is not true", OperationKindEx.NegatedPattern, "False", "True")]
+        [DataRow("nullableBoolNull", "is not true", OperationKindEx.NegatedPattern, null, "True")]
+        [DataRow("nullableBoolNull", "is not false", OperationKindEx.NegatedPattern, null, "False")]
```

Comment:
The `nullableBool`s above should only have a single state. Leave a "Should be" comment.